### PR TITLE
crimson/test: comment out EC test cases and some cases not supported now

### PR DESCRIPTION
**Pull request by liu-chunmei**
Signed-off-by: chunmei-liu <chunmei.liu@intel.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by athanatos** [comment]
Hmm, this isn't actually a very good solution.  We're going to want to be able to generate a single build with both classic and crimson eventually, and if-defing out portions of client-side testing won't work well.  Instead, I think we probably want to check for a CRIMSON_COMPAT environment variable and skip tests dynamically on that basis.  @rzarzynski Does that make sense to you?

**Comment by liu-chunmei** [comment]
> Hmm, this isn't actually a very good solution. We're going to want to be able to generate a single build with both classic and crimson eventually, and if-defing out portions of client-side testing won't work well. Instead, I think we probably want to check for a CRIMSON_COMPAT environment variable and skip tests dynamically on that basis. We can modify the teuthology task to set that env variable for crimson clusters. @rzarzynski Does that make sense to you?

teuthology setup the environment variable,  but each test case should check this env variable to decide to run or not, right? 
such as:

**Comment by liu-chunmei** [comment]
> Hmm, this isn't actually a very good solution. We're going to want to be able to generate a single build with both classic and crimson eventually, and if-defing out portions of client-side testing won't work well. Instead, I think we probably want to check for a CRIMSON_COMPAT environment variable and skip tests dynamically on that basis. We can modify the teuthology task to set that env variable for crimson clusters. @rzarzynski Does that make sense to you?

check environment variable now.

File: src/test/librados/crimson_env.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <cstdlib>
+
+#define CHECK_ENV();                    \
+  if (getenv("CRIMSON_COMPAT")) {       \
+    return;                             \
```
Comment (rzarzynski) [inline src/test/librados/crimson_env.h:10]: I don't think returning silently is a good. We should be able to differentiate between _green_ and _skipped_ tests. IIUC now everything would be green.

How about [`GTEST_SKIP()`](http://google.github.io/googletest/advanced.html#skipping-test-execution)?

Anyway, it's just a detail. I like the env. var. idea.

Comment (liu-chunmei) [reply src/test/librados/crimson_env.h]: changed to use GTEST_SKIP().

File: src/test/librados/crimson_env.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <cstdlib>
+
+#define CHECK_ENV();                    			\
+  if (getenv("CRIMSON_COMPAT")) {  			     	\
+    GTEST_SKIP() << "Not supported, Skipping single test";	\
+  }
```
Comment (athanatos) [inline src/test/librados/crimson_env.h:11]: Let's name this SKIP_IF_CRIMSON()

Let's also introduce a helper is_crimson_cluster() for...

File: src/test/librados/list.cc [hunk 3]
```diff
@@ -433,9 +438,10 @@ TEST_F(LibRadosList, EnumerateObjects) {
 
   // Ensure a non-power-of-two PG count to avoid only
   // touching the easy path.
-  ASSERT_TRUE(set_pg_num(&s_cluster, pool_name, 11).empty());
-  ASSERT_TRUE(set_pgp_num(&s_cluster, pool_name, 11).empty());
-
+  if (!getenv("CRIMSON_COMPAT")) {
```
Comment (athanatos) [inline src/test/librados/list.cc:441]: Let's use the is_crimson_cluster() here -- we only want to reference CRIMSON_COMPAT in the header crimson_env.h header you created.

**Comment by athanatos** [comment]
Looks about right other than the naming nit.  This does need to go through the normal ceph teuthology suite prior to merge since it touches things outside of crimson/

File: src/test/librados/crimson_env.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+bool is_crimson_cluster() {
+  if (getenv("CRIMSON_COMPAT")) {
+    return true;
+  } else {
+    return false;
+  }
+}
```
Comment (tchaikov) [inline src/test/librados/crimson_env.h:19]:
```suggestion
bool is_crimson_cluster() {
  return (getenv("CRIMSON_COMPAT") != nullptr);
}

#define SKIP_IF_CRIMSON()                           \
  if (is_crimson_cluster()) {                       \
    GTEST_SKIP() << "Not supported by crimson yet. Skipped"; \
  }


```

**Comment by tchaikov** [comment]
sorry, i just submit a commit with my proposed change. i misinterpreted the prompt in the UI (my first time using the GitHub app to read this in Chinese), thought it was pending. please feel free to drop it if it does not make sense to you. but i still think it does for couple reasons

- it fixes the build by removing the comma after `SKIP_IF_CRIMSON();`
- it reuse the helper of `is_crimson_cluster()`. less repeatings this way.
- it avoids the manual dispatch for returning `true` or `false`
- it highlights the reason why the test is skipped.

**Comment by liu-chunmei** [comment]
> sorry, i just submitted a commit with my proposed change. i misinterpreted the prompt in the UI (my first time using the GitHub app to read this in Chinese), thought it was pending. please feel free to drop it if it does not make sense to you. but i still think it does for couple reasons
> 
> * it fixes the build by removing the comma after `SKIP_IF_CRIMSON();`
> * it reuse the helper of `is_crimson_cluster()`. less repeatings this way.
> * it avoids the manual dispatch for returning `true` or `false`
> * it highlights the reason why the test is skipped.

@tchaikov LGTM. you can just rename the header file and push it again.

**Comment by tchaikov** [comment]
@liu-chunmei i'd rather leave this to the author of a commit.

File: src/test/librados/crimson_utils.h [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <cstdlib>
+
+bool is_crimson_cluster();
```
Comment (tchaikov) [inline src/test/librados/crimson_utils.h:8]: not sure if this tiny function deserves a .cc file, how about
```c++
static inline bool is_crimson_cluster() {
  return getenv("CRIMSON_COMPAT") != nullptr;
}
```

also, i'd suggest drop the parenthesis around the expression.

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
apart from the nit, LGTM.

File: src/test/librados/crimson_utils.h (hunk 5 above)
Comment (liu-chunmei) [reply src/test/librados/crimson_utils.h]: done.

**Review by tchaikov — APPROVED** [review]

**Pull request merged**